src/client/gui/ffi/dart_ffi.cpp

```diff
@@ -214,4 +215,23 @@ catch (...)
     mpl::log(mpl::Level::warning, category, "failed converting memory to bytes");
     return -1;
 }
+
+const char* default_mount_target(const char* source)
+try
+{
+    const QString q_source{source};
+    free((void*)source);
+    const auto target = MP_UTILS.default_mount_target(q_source).toStdString();
+    return strdup(target.c_str());
+}
+catch (const std::exception& e)
+{
+    mpl::log(mpl::Level::warning, category, fmt::format("failed retrieving default mount target: {}", e.what()));
+    return nullptr;
+}
+catch (...)
+{
+    mpl::log(mpl::Level::warning, category, "failed retrieving default mount target");
```

Comment:
Could you please factor out the error message? Can't use function try any longer, but that should be fine. You can have a `static constexpr char *` at the beginning.

Reply:
Yes, sure. It just sucks that it adds another level of indentation for the whole body of the function...